eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -167,6 +167,34 @@
         }
       }
     },
+    {
+      "StaticConfigs": {
+        "Ubuntu22": {
+          "OSVmImage": "MMSUbuntu22.04",
+          "Pool": "azsdk-pool-mms-ubuntu-2204-general",
+          "VCPKG_DEFAULT_TRIPLET": "x64-linux",
+          "BuildArgs": "-j 10",
+          "RunProxyTests": true
+        }
+      },
+      "BuildSettings": {
+        "clang-13": {
+          "AptDependencies": "clang-13",
+          "CC": "/usr/bin/clang-13",
+          "CXX": "/usr/bin/clang++-13"
+        },
+        "clang-15": {
+          "AptDependencies": "clang-15",
+          "CC": "/usr/bin/clang-15",
+          "CXX": "/usr/bin/clang++-15"
+        },
+        "included_debug": {
+          "CMAKE_BUILD_TYPE": "Debug",
+          "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON",
```

Comment:
Should we be building samples too?

Reply:
Maybe? My primary focus for this PR is vcpkg, which doesn't build samples (or tests).